pkg/types/baremetal/platform.go

```diff
@@ -106,4 +106,10 @@ type Platform struct {
 	//
 	// +optional
 	ClusterOSImage string `json:"clusterOSImage,omitempty" validate:"omitempty,osimageuri,urlexist"`
+
+	// SkipValidations is list of validations to skip during loading/fetching of install-config.yaml for baremetal platform
+	// currently to validations skipping is supported
+	// "interfaces" - skip validation of hosts networking interfaces via libvirt
+	// "hosts" - skips validation of hosts existannce it they values
+	SkipValidations string `json:"skipValidations,omitemtpy"`
```

Comment:
Did you look into alternatives that avoid this new boolean?

I think the main issue here is that the validations happen on `create ignition-configs` - I wonder if we can either ensure they only happen on `create cluster` or alternatively use some special value to `LibvirtURI` that indicates libvirt related validations should be disabled?

/cc @stbenjam any thoughts on the best interface here?

Reply:
I agree -- adding anything to the platform is a last resort.  Changing when the validations run may be the best solution.

If we need to run some validations before create ignition to make sure we don't make bad ignition configs, I'd prefer to opt for some indication we're doing an assisted install. Disabling validations is probably not the only conditional thing we'll need to do for it.

Reply:
This is another example of why I think "assisted-baremetal" is a different platform in the installer, even if the platform in the cluster is still just "baremetal".

Reply:
@dhellmann if IPI was optional, those checks would not be required. Also, I would expect the runtime validations (libvirt socket) to be done during create cluster and not generate igntions.

Reply:
@hardys i wanted to disrupt the usual flow as little as possible, so for user that do not add the flag to the install-config, nothing would change. I had 2 addtional options that i thought about
1) change the build tag of pkg/types/baremetal/validation/libvirt.go  from baremetal to libvirt, and compile openshift-installer with TAG="baremetal" instead of TAG="baremetal libvirt". This would remove libvirt validation of network interfaces from baremetal installer.
2) make Hosts field in pkg/types/baremetal/platform.go optional, and then execute all the validations for Hosts only in case this field is not empty

Reply:
> @dhellmann if IPI was optional, those checks would not be required. Also, I would expect the runtime validations (libvirt socket) to be done during create cluster and not generate igntions.

During an IPI deployment, things like the BMC credentials are required. During an assisted installer deployment they aren't. We know in advance which things are required, right? So instead of adding a field that's a list of validations to skip, we could use 1 setting to tell us which mode we're using and let the code perform or skip the appropriate validations. That feels a lot like how the platform value is currently used.

Reply:
Using a different platform will also allow the BareMetalHost resources to be configured as externally provisioned (the control plane nodes are now, but the workers are not).

Reply:
We are going to push back against any new field added to the platform. It's not that we're trying to be difficult, but because we're concerned about the user experience of an already complicated setup. Every field we add is another setting that needs documenting and users will see it and need to consider it (even if they don't need to do anything with it explicitly).

If we can bolt assisted installer onto the side of the `baremetal` IPI platform without adding complexity then we are happy with that. Adding knobs isn't a viable solution.

I like the suggestion of reworking the validations such that runtime validations happen at `create cluster` time.

Reply:
I don't think assisted install should just turn off all host validations, we want BareMetalHost records for the control plane -- they just lack power credentials, right?

Reply:
> I don't think assisted install should just turn off all host validations, we want BareMetalHost records for the control plane -- they just lack BMC credentials, right?

Yes, we want them. They will need a name, MAC address, and role.

Reply:
suppose that for now we will leave BMH validation as is ( it does not really hurt us, we will just fill in dummy values, and we will use LibvirtURI field in baremetal platform to signal if it is assisted-installer setup. the code will look like:
if p.LibvirtURI == "assisted-installer" {
     skip libvirt interface validation
}

@stbenjam @hardys

Reply:
Magic string values aren't ok here -- I think the consensus from Steve and I are that the IPI-specific validations should be moved to be later, so you can do the UPI workflow with the baremetal platform.

Reply:
If you are just looking to iterate quickly, then I'd suggest doing it in a branch/fork. That worked well for us early on in baremetal IPI days, because we could iterate really quickly without the scrutiny that comes with merging to the installer master branch.

Reply:
> I like the suggestion of reworking the validations such that runtime validations happen at `create cluster` time.

@stbenjam do you think thats something someone external to the team can develop? I'm concerned this is not a trivial effort and implies only long term solution? 

> We are going to push back against any new field added to the platform. It's not that we're trying to be difficult, but because we're concerned about the user experience of an already complicated setup. Every field we add is another setting that needs documenting and users will see it and need to consider it (even if they don't need to do anything with it explicitly).

do we see a path for an internal only flag? e.g. not supported / documented outside this usage path?

Reply:
Yes, we can help - @dhellmann is going to put together a proposal to make sure we understand exactly what you need.

https://github.com/openshift/installer/pull/3794#issuecomment-649034526